web/src/settings_banner.ts

```diff
@@ -0,0 +1,163 @@
+    }),
+    buttons: [
+        {
+            label: $t({defaultMessage: "Learn more"}),
+            custom_classes: "user-groups-info",
+            attention: "quiet",
+        },
+    ],
+    close_button: false,
+};
+
+const STREAM_INFO_BANNER: Banner = {
+    intent: "info",
+    label: $t({
+        defaultMessage: "Channels organize conversations based on who needs to see them.",
+    }),
+    buttons: [
+        {
+            label: $t({defaultMessage: "Learn more"}),
+            custom_classes: "stream-info",
+            attention: "quiet",
+        },
+    ],
+    close_button: false,
+};
+
+const MOBILE_PUSH_NOTIFICATION_BANNER: Banner = {
+    intent: "warning",
+    label: $t({
+        defaultMessage: "Mobile push notifications are not enabled on this server.",
+    }),
+    buttons: [
+        {
+            label: $t({defaultMessage: "Learn more"}),
+            custom_classes: "mobile-push-notification-info",
+            attention: "quiet",
+        },
+    ],
+    custom_classes: "mobile-push-notifications-banner",
+    close_button: false,
+};
+
+export function set_up_upgrade_banners(): void {
+    const has_billing_access = settings_data.user_has_billing_access();
+    const is_business_type_org =
+        realm.realm_org_type === settings_config.all_org_type_values.business.code;
+    const $upgrade_container = $(".upgrade-organization-banner-container");
+
+    if ($upgrade_container.length === 0) {
+        return;
+    }
+
+    let banner;
+    if (is_business_type_org) {
+        banner = has_billing_access ? UPGRADE_ACCESS_BANNER : UPGRADE_TO_ZULIP_CLOUD_BANNER;
+    } else {
+        banner = has_billing_access ? UPGRADE_OR_SPONSORSHIP_BANNER : UPGRADE_TO_ZULIP_CLOUD_BANNER;
+    }
+    banners.open(banner, $upgrade_container);
+}
+
+export function set_up_group_info_banner(): void {
+    const $container = $(".group-info-banner");
+    if ($container.length === 0) {
+        return;
+    }
+
+    banners.open(GROUP_INFO_BANNER, $container);
+
+    $container.on("click", ".user-groups-info", (e) => {
+        e.preventDefault();
+        window.open("/help/user-groups", "_blank", "noopener,noreferrer");
+    });
+}
+
+export function set_up_stream_info_banner(): void {
+    const $container = $(".stream-info-banner");
+    if ($container.length === 0) {
+        return;
+    }
+
+    banners.open(STREAM_INFO_BANNER, $container);
+
+    $container.on("click", ".stream-info", (e) => {
+        e.preventDefault();
+        window.open("/help/introduction-to-channels", "_blank", "noopener,noreferrer");
+    });
+}
+
+export function set_up_mobile_push_banner(): void {
+    const $container = $(".mobile-push-notifications-banner-container");
+    if ($container.length === 0) {
+        return;
+    }
+
+    banners.open(MOBILE_PUSH_NOTIFICATION_BANNER, $container);
+
+    $container.on("click", ".mobile-push-notification-info", (e) => {
```

Comment:
The link is opened twice if I click on the button after updating one of the settings in the table.